syntax = "proto3";

package protodb.index;

import "protodb/index/index.proto";

message GetIndexRequest {
    string database = 1;
    string collection = 2;
    uint64 id = 3;
}

message GetIndexResponse {
    enum ErrorCode {
        NO_ERROR = 0;
        INTERNAL_ERROR = 1;
        INVALID_DATABASE = 2;
        INVALID_COLLECTION = 3;
        INVALID_ID = 4;
    }
    ErrorCode error_code = 1;

    Index index = 2;
}
